quartz/plugins/emitters/offline.ts

```diff
@@ -0,0 +1,54 @@
+import { QuartzEmitterPlugin } from "../types"
+import { FilePath, FullSlug } from "../../util/path"
+import path from "path"
+import { readFileSync } from "node:fs"
+
+export const Offline: QuartzEmitterPlugin = () => {
+  return {
+    name: "Offline",
+    getQuartzComponents() {
+      return []
+    },
+    async emit({ cfg }, _content, _resources, emit): Promise<FilePath[]> {
+      const manifest = {
+        short_name: cfg.configuration.pageTitle,
+        name: cfg.configuration.pageTitle,
+        description: cfg.configuration.description,
+        background_color: cfg.configuration.theme.colors.lightMode.light,
+        theme_color: cfg.configuration.theme.colors.lightMode.light,
+        display: "minimal-ui",
+        icons: [
+          {
+            src: "static/icon.svg",
+            sizes: "any",
+            purpose: "maskable",
+          },
+          {
+            src: "static/icon.svg",
+            sizes: "any",
+            purpose: "any",
+          },
+        ],
+        start_url: "/",
+      }
+
+      const serviceWorker =
+        "importScripts('https://storage.googleapis.com/workbox-cdn/releases/6.4.1/workbox-sw.js');" +
+        "const {pageCache, imageCache, staticResourceCache, googleFontsCache, offlineFallback} = workbox.recipes;" +
+        "pageCache(); googleFontsCache(); staticResourceCache(); imageCache();offlineFallback();"
```

Comment:
can we use string interpolation here?

Reply:
Sure, it'll just be more bytes,

Reply:
i prefer string interpolation for readability reasons